tests/suites/helpers.function

```diff
@@ -423,17 +461,11 @@ jmp_buf jmp_tmp;
 /*----------------------------------------------------------------------------*/
 /* Helper Functions */
 
-void test_fail( const char *test, int line_no, const char* filename )
```

Comment:
Instead of deleting `test_fail`, we could have it be a static function that calls `set_test_result()`

Reply:
initially `set_test_result()` was static, but this caused compilation warnings, as declared functions that are not used, in modules that are configured out. The linker probably doesn't remove static functions.
In addition, since I thought putting one single function would make it cleaner